### PR TITLE
[WIP] Types

### PR DESCRIPTION
**Pull request by CallMeFoxie**
DO NOT MERGE!

Write notes about missing types before we merge this! Also squash before merging.

**Reffers:** #15 #16 #8 #6 #5

### CODE REVIEW

**Comment by purrplingcat** [comment]
Pls target related issues in commits in this PR.

File: types/types.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package types
+
+type Device struct {
+	Title          string
+	Description    string
+	Location       string
```
Comment (purrplingcat) [inline types/types.go:10]: Please specify location types for GPS coords, City and/or Locator

File: types/types.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+	FeaturedSensor *Sensor
+	Tags           []string
+}
+
+type Sensor struct {
+	Alias       string
+	Title       string
+	Type        int
```
Comment (purrplingcat) [inline types/types.go]: Maybe use enum instead of int?

File: types/types.go [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package types
+
+type Device struct {
+	Title          string
+	Description    string
+	Location       string
+	FeaturedSensor *Sensor
+	Tags           []string
```
Comment (purrplingcat) [inline types/types.go:12]: Create type Tag instead of string[] and save tags to db as standalone document? Better for tag cloud and searching. Or not?

**Comment by CallMeFoxie** [comment]
cc #15 #16 #8 #6 #5

**Pull request merged**